src/AutoRest.CSharp.V3/ClientModels/ClientBuilder.cs

```diff
@@ -185,16 +182,41 @@ private static ServiceClientParameter[] OrderParameters(IEnumerable<ServiceClien
             );
 
             string operationName = operation.CSharpName();
+            //TODO: This is a hack since we don't have the model information at this point
+            var schemaForPaging = ((responseBody as ObjectResponseBody)?.Type as SchemaTypeReference)?.Schema as ObjectSchema;
             return new ClientMethod(
                 operationName,
                 ClientModelBuilderHelpers.EscapeXmlDescription(operation.Language.Default.Description),
                 request,
                 OrderParameters(methodParameters),
                 clientResponse,
-                new ClientMethodDiagnostics($"{clientName}.{operationName}", Array.Empty<DiagnosticScopeAttributes>())
+                new ClientMethodDiagnostics($"{clientName}.{operationName}",Array.Empty<DiagnosticScopeAttributes>()),
+                GetClientMethodPaging(operation, schemaForPaging)
             );
         }
 
+        private static ClientMethodPaging? GetClientMethodPaging(Operation operation, ObjectSchema? schema)
```

Comment:
Hm, wonder if we need something like `SchemaPropertyReference` that would be similar to schema type reference but with a property.